pkg/alertmanager/alertmanager_ring.go

```diff
@@ -32,6 +32,12 @@ var RingOp = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, func(s ring.InstanceS
 	return s != ring.ACTIVE
 })
 
+// UserOwnedRingOp is the operation used for checking if a user is owned by an alertmanager.
+var UserOwnedRingOp = ring.NewOp([]ring.InstanceState{ring.ACTIVE, ring.JOINING}, func(s ring.InstanceState) bool {
+	// A user is owned by an alertmanager in both ACTIVE and JOINING states.
+	return (s != ring.ACTIVE && s != ring.JOINING)
```

Comment:
I'm wondering if this is correct or it should stay to `ACTIVE`. Look at this in the store-gateway:
https://github.com/cortexproject/cortex/blob/d3068f97b38a67d4c973edc404d450dc5be71d11/pkg/storegateway/gateway_ring.go#L35-L41

The idea was that while joining (eg. because scaling up), the previous 3 replicas for a given tenant shouldn't be reduced to 2. If `JOINING` doesn't extend the replica set, then 1 out of the 3 previous replicas offloads the tenant before this new instance is `ACTIVE`.